okhttp/src/main/java/okhttp3/internal/cache/DiskLruCache.java

```diff
@@ -753,6 +753,8 @@ private void validateKey(String key) {
 
           while (delegate.hasNext()) {
             Entry entry = delegate.next();
+            if (entry == null || !entry.readable) continue; // Entry during edit
```

Comment:
Can entry ever be null here?

Is it simpler to move this check inside of entry.snapshot()?

Reply:
Not sure if entry can be ever null. I don't know the internal assumptions of OkHttp and based my code on line 42 in Snapshot get(String key) method, where same check is being done.